framework/logging.cpp

```diff
@@ -1545,43 +1597,86 @@ static void print_content_single_line(int fd, std::string_view before,
     writeln(fd, before, escape_for_single_line(message, escaped), after);
 }
 
-static void format_and_print_message(int fd, int message_level, std::string_view message)
+static void format_and_print_message(int fd, int message_level, std::string_view message, bool from_thread_message)
 {
-    const char *levels[] = { "error", "warning", "info", "debug" }; //levels for yaml
+    const char *levels[] = { "error", "warning", "info", "debug", "skip" }; //levels for yaml
 
     if (message.find('\n') != std::string_view::npos) {
         /* multi line */
         if (current_output_format() == SandstoneApplication::OutputFormat::yaml) {
-            if (message_level > int(std::size(levels)))
-                message_level = std::size(levels) - 1;
-
-            // trim a trailing newline, if any (just one)
-            if (message[message.size() - 1] == '\n')
-                message.remove_suffix(1);
-            writeln(fd, indent_spaces(), "    - level: ", levels[message_level]);
-            writeln(fd, indent_spaces(), "      text: |1");
-            print_content_indented(fd, "       ", message);
+            if (from_thread_message) {
+                 if (message_level > int(std::size(levels)))
+                    message_level = std::size(levels) - 1;
+
+                // trim a trailing newline, if any (just one)
+                if (message[message.size() - 1] == '\n')
+                    message.remove_suffix(1);
+
+                writeln(fd, indent_spaces(), "    - level: ", levels[message_level]);
+                writeln(fd, indent_spaces(), "      text: |1");
+                print_content_indented(fd, "       ", message);
+            } else {
+                writeln(fd, indent_spaces(), "  skip-reason: |1");
+                print_content_indented(fd, "   ", message);
+            }   
         } else {
-            writeln(fd, "  - |");
+            if (from_thread_message)
+                writeln(fd, "  - |");
+            else
+                writeln(fd, indent_spaces(), "\n  - |");
```

Comment:
`indent_spaces()` followed by a newline sounds wrong. Did you mean `"\n", indent_spaces(), "  - |"`?

Reply:
Yes, corrected this.